File: Practica5/Marco teorico/3. Destructor/B. Declaracion de destructores/3.2.cpp
#include <iostream>
#include <string>


using namespace std;


class MiClase {
private:
    string nombre;


public:
    // Constructor
    MiClase(string n) : nombre(n) {
        cout << "Constructor llamado para " << nombre << endl;
    }


    // Destructor
    ~MiClase() {
        cout << "Destructor llamado para " << nombre << endl;
    }


    // Método para mostrar el nombre
    void mostrarNombre() {
        cout << "Nombre: " << nombre << endl;
    }
};

int main() {
    // Crear un objeto de la clase MiClase
    MiClase objeto("Ejemplo");


    // Llamar al método para mostrar el nombre
    objeto.mostrarNombre();


    return 0;
}

File: Practica5/Marco teorico/3. Destructor/B. Declaracion de destructores/3.2.java
public class MiClase {
    private String nombre;

    public MiClase(String n) {
        nombre = n;
        System.out.println("Constructor llamado para " + nombre);
    }

    public void mostrarNombre() {
        System.out.println("Nombre: " + nombre);
    }

}

public class Main {
    public static void main(String[] args) {
        {
  
            MiClase objeto = new MiClase("Objeto 1");

            objeto.mostrarNombre();
        }
    }
}
